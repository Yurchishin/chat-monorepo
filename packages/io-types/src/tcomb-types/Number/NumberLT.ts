import t from 'tcomb-validation'
import R from '@chat/ramda'

export const NumberLT = (max: number) => t.refinement(
    t.Number,
    R.lt(R.__, max),
    'NumberLT',
)
