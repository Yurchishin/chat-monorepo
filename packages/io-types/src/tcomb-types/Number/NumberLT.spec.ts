import {NumberLT} from './NumberLT'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('NumberLT(5)', () => {
	const LT5 = NumberLT(5)

	expectTypeSuccess(LT5, -1)
	expectTypeFailure(LT5, 5)
	expectTypeFailure(LT5, 6)
	expectTypeFailure(LT5, '1')
})
