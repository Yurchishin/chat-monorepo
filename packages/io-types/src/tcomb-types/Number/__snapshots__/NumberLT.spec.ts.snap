// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberLT(5) expectTypeFailure(1): TYPE_FAILURE::NumberLT(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "1",
      "expected": [Function],
      "message": "Invalid value \\"1\\" supplied to Number",
      "path": Array [],
    },
  ],
  "value": "1",
}
`;

exports[`NumberLT(5) expectTypeFailure(5): TYPE_FAILURE::NumberLT(5) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 5,
      "expected": [Function],
      "message": "Invalid value 5 supplied to NumberLT",
      "path": Array [],
    },
  ],
  "value": 5,
}
`;

exports[`NumberLT(5) expectTypeFailure(6): TYPE_FAILURE::NumberLT(6) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 6,
      "expected": [Function],
      "message": "Invalid value 6 supplied to NumberLT",
      "path": Array [],
    },
  ],
  "value": 6,
}
`;
